PORTAL/jobs/requests.py

```diff
@@ -320,41 +346,42 @@ def __init__(self, reqid, reqdir, status=STATUS.CREATED, history=None):
             history=history,
         )
 
+        self._request: Optional[Request] = None
+        self._get_request: Optional[Callable[(str, str), Request]] = None
+        self._fs = Optional[_utils.FSTree] = None
+
     def __str__(self):
         return str(self.reqid)
 
     @property
-    def short(self):
+    def short(self) -> str:
         if not self.status:
             return f'<{self.reqid}: (created)>'
         return f'<{self.reqid}: {self.status}>'
 
     @property
-    def request(self):
-        try:
-            return self._request
-        except AttributeError:
+    def request(self) -> Request:
+        if self._request is None:
             get_request = getattr(self, '_get_request', Request)
             self._request = get_request(self.reqid, self.reqdir)
-            return self._request
+        return self._request
 
     @property
-    def fs(self):
-        try:
-            return self._fs
-        except AttributeError:
+    def fs(self) -> _utils.FSTree:
+        if self._fs is None:
             raise NotImplementedError
+        return self._fs
 
     @property
-    def host(self):
+    def host(self) -> str:
         # XXX This will need to support other hosts.
         return 'fc_linux'
 
     @property
-    def started(self):
+    def started(self) -> Tuple[Optional[datetime.datetime], Optional[str]]:
```

Comment:
If `Result.STATUS` were an enum...

```suggestion
    def started(self) -> Tuple[Optional[datetime.datetime], Optional[Result.STATUS]]:
```

Reply:
Also, (I think) we can be more clear about the optional results:

```suggestion
    def started(self) -> Union[Tuple[datetime.datetime], str], Tuple[None, None]]:
```

Reply:
Yeah, no enums or even enum-like-things (yet), but the `Union` suggestion works.